File: output/gemma4-e2b_IO_1/avatar/Java/Python/codeforces_306_A.py
import sys

def solve():
    # Read input from stdin
    try:
        # Read n and m from the input line
        data = sys.stdin.read().split()
        if len(data) < 2:
            return
        n = int(data[0])
        m = int(data[1])
    except Exception:
        # Handle case where input might be missing or invalid, though for competitive programming context, we assume valid input structure.
        return

    # Initialize the array a of size m
    a = [0] * m

    if n % m == 0:
        # Case 1: n is divisible by m
        result = n // m
        for i in range(m):
            a[i] = result
    else:
        # Case 2: n is not divisible by m
        sub = n // m
        for i in range(m):
            a[i] = sub
        
        test = n - (sub * m)
        count = 0
        
        # Increment the first 'test' elements by 1, wrapping around
        for i in range(test):
            a[count] += 1
            count = (count + 1) % m

    # Output the elements separated by spaces
    print(*(a))

File: output/gemma4-e2b/avatar/Python/Java/codeforces_306_A.java
import java.util.Scanner;

public class codeforces_306_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read n and m from a single line of input
        if (scanner.hasNextInt()) {
            int n = scanner.nextInt();
            if (scanner.hasNextInt()) {
                int m = scanner.nextInt();

                int q = n / m;
                int r = n % m;

                int[] a = new int[m];
                
                // Initialize all elements to q
                for (int i = 0; i < m; i++) {
                    a[i] = q;
                }

                // Add 1 to the first r elements
                for (int i = 0; i < r; i++) {
                    a[i] += 1;
                }

                // Print the result
                for (int i = 0; i < m; i++) {
                    System.out.print(a[i] + " ");
                }
                System.out.println();
            }
        }
        scanner.close();
    }
}